modules/system-disaster-recovery/src/main/java/org/apache/ignite/internal/disaster/system/SystemDisasterRecoveryManagerImpl.java

```diff
@@ -452,6 +465,27 @@ public CompletableFuture<Void> migrate(ClusterState targetClusterState) {
                 }, restartExecutor);
     }
 
+    private static boolean isDescendantOrSame(ClusterState potencialDescendant, ClusterState potentialAncestor) {
+        List<UUID> descendantLineage = extractClusterIdsLineage(potencialDescendant);
+        List<UUID> ancestorLineage = extractClusterIdsLineage(potentialAncestor);
+
+        return descendantLineage.size() >= ancestorLineage.size()
+                && descendantLineage.subList(0, ancestorLineage.size()).equals(ancestorLineage);
+    }
+
+    private static List<UUID> extractClusterIdsLineage(ClusterState state) {
+        List<UUID> lineage = new ArrayList<>();
```

Comment:
```suggestion
        var lineage = new ArrayList<UUID>();
```

Reply:
I prefer leaving it as is (the difference is minor, and a full-blown rerun of tests is not worth it, I think)